dace/transformation/dataflow/prune_connectors.py

```diff
@@ -153,22 +153,40 @@ def _candidates(nsdfg: nodes.NestedSDFG) -> Set[str]:
 
         return candidates
 
+    def _find_symbols_that_can_not_be_removed(self, sdfg: SDFG) -> Set[str]:
+        """Find the set of symbols that can not be removed."""
+
+        # The implementation of this function is such that it has the same behaviour as before,
+        #  where `dace.transformation.helpers.is_symbol_unused()` was used. However, I am sure
+        #  that it can be implemented in terms of `used_symbols()` but I am unsure if it is
+        #  so simple as `return sdfg.used_symbols()`?
+
+        unremovable_symbols: Set[str] = set()
+
+        for desc in sdfg.arrays.values():
+            unremovable_symbols.update(map(str, desc.free_symbols))
+        for state in sdfg.states():  # Previously `nodes()` was used.
```

Comment:
```suggestion
        for state in sdfg.states():
```

`nodes()` was definitely wrong, if operating on a per-SDFG level as opposed to a per-CFG level - good catch.